clusterman/aws/auto_scaling_resource_group.py

```diff
@@ -90,18 +90,17 @@ def _launch_config(self) -> Dict[str, Any]:
     def market_weight(self, market: InstanceMarket) -> float:
         """ Returns the weight of a given market
 
-        ASGs have no concept of weight, so if ASG is available in the market's
-        AZ, we return 1 for that market.
+        ASGs can be defined with different instance weights. If we can find
+        the weight for a given instance type, we return it. Otherwise we
+        default to 1.
 
         :param market: The market for which we want the weight for
         :returns: The weight of a given market
         """
         if market.az in self._group_config['AvailabilityZones']:
-            if market.instance != self._launch_config['InstanceType']:
-                logger.warning(
-                    f'Instance type {market.instance} is different from launch '
-                    f'config instance type: {self._launch_config["InstanceType"]}'
-                )
+            for instance in self._group_config.get('Instances', []):
+                if market.instance == instance.get('InstanceType'):
+                    return int(instance.get('WeightedCapacity', '1'))
```

Comment:
Is the weight constrained to integers?  For spot fleets, we use fractional weights.

Reply:
Hmm good point...
The [autoscaling API](https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_LaunchTemplateOverrides.html) expects a string value that matches the following pattern: `^[1-9][0-9]{0,2}$`
So it's just integers as far as I can tell.